docs/Troubleshooting.md

```diff
@@ -0,0 +1,177 @@
+* Double check that the following OMS Service endpoints are whitelisted
+
+Agent Resource | Ports 
+---- | ----
+*.ods.opinsights.azure.com | Port 443
+*.oms.opinsights.azure.com | Port 443
+ods.systemcenteradvisor.com | Port 443
+*.blob.core.windows.net/ | Port 443
+
+### I'm getting a 403 when I'm trying to onboard!
+#### Probable Causes
+* Date and Time is incorrect on Linux Server
+* Workspace ID and Workspace Key used are not correct
+
+#### Resolution
+* Check the time on your Linux server with the command `date`. if the data is +/- 15 minutes from current time then onboarding fails. To correct this update, the date and/or timezone of your Linux server.
+ * **New!** The latest version of the OMS Agent for Linux now notifies you if the time skew is causing the onboarding failure
+* Re-onboard using correct Workspace ID and Workspace Key [instructions](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/docs/OMS-Agent-for-Linux.md#onboarding-using-the-command-line)
+
+### I'm seeing a 500 Error and 404 Error in the log file right after onboarding
+This is a known issue an occurs on first upload of Linux data into an OMS workspace. This does not affect data being sent or service experience.
+
+### My Nagios data is not showing up in the OMS Portal!!
+#### Probable Causes
+* omsagent user does not have permissions to read from Nagios log file
+* Nagios source and filter have not been uncommented from omsagent.conf file
+
+#### Resolutions
+* Add omsagent user to read from Nagios file [instructions](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/docs/OMS-Agent-for-Linux.md#nagios-alerts)
+* In the OMS Agent for Linux general configuration file at `/etc/opt/microsoft/omsagent/conf/omsagent.conf` ensure that **both** the Nagios source and filter are uncommented
+```
+<source>
+  type tail
+  path /var/log/nagios/nagios.log
+  format none
+  tag oms.nagios
+</source>
+
+<filter oms.nagios>
+  type filter_nagios_log
+</filter>
+```
+
+### I'm not seeing any Linux data in the OMS Portal
+#### Probable Causes
+* Onboarding to the OMS Service failed
+* Connection to the OMS Service is blocked
+* OMS Agent for Linux data is backed up
+
+### Resolutions
+* Check if onboarding the OMS Service was successful by checking if the following file exists: `/etc/opt/microsoft/omsagent/conf/omsadmin.conf`
+ * Re-onboard using the omsadmin.sh command line [instructions](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/docs/OMS-Agent-for-Linux.md#onboarding-using-the-command-line)
+* If using a proxy, check proxy troubleshooting steps aboce
+* In some cases, when the OMS Agent for Linux cannot talk to the OMS Service, data on the Agent is backed up to the full buffer size: 50 MB. The OMS Agent for Linux should be restarted by running the following command `service omsagent restart` or `systemctl restart omsagent`.
+ * **Note:** This issue is being remedied for the next Generally Available OMS Agent for Linux
```

Comment:
That's really confusing, and dated.

How about something like "This issue is resolved will be available in the next release. Versions <= x.y.z are susceptible to this issue.". That way, you can check your version and know if you're good or not.